File: checker/src/utils/response.ts
const CORS_HEADERS = {
  "Content-Type": "application/json",
  "Access-Control-Allow-Origin": "*",
  "Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
  "Access-Control-Allow-Headers": "Content-Type, Authorization",
  "Access-Control-Max-Age": "86400",
};

export function corsResponse(status = 204): Response {
  return new Response(null, {
    status,
    headers: CORS_HEADERS,
  });
}

export function jsonResponse(data: any, status = 200): Response {
  return new Response(JSON.stringify(data), {
    status,
    headers: CORS_HEADERS,
  });
}

export function successResponse(data: any, message = "Success", status = 200): Response {
  return jsonResponse({
    success: true,
    message,
    data,
  }, status);
}

export function errorResponse(message: string, status = 400, error?: any): Response {
  return jsonResponse({
    success: false,
    message,
    error: error || null,
  }, status);
}

export function paginatedResponse(data: any[], total: number, page: number, limit: number, status = 200): Response {
  const totalPages = Math.ceil(total / limit);
  return jsonResponse({
    success: true,
    data,
    pagination: {
      page,
      limit,
      total,
      totalPages,
    },
  }, status);
}

File: microservice-checker/domain-service/internal/shared/dto/response.go
package dto

type APIResponse struct {
	Success bool        `json:"success"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
	Errors  []string    `json:"errors,omitempty"`
	Error   interface{} `json:"error"`
}

type PaginationInfo struct {
	Page       int   `json:"page"`
	Limit      int   `json:"limit"`
	Total      int64 `json:"total"`
	TotalPages int   `json:"totalPages"`
}

type PaginatedResponse struct {
	Success    bool           `json:"success"`
	Data       interface{}    `json:"data"`
	Pagination PaginationInfo `json:"pagination"`
}
